mergin/client_pull.py

```diff
@@ -551,3 +572,176 @@ def pull_project_finalize(job):
 
     shutil.rmtree(job.temp_dir)
     return conflicts
+
+
+def download_file_async(mc, project_path, file_path, output_file, version):
+    """
+    Starts background download project file at specified versions.
+    Returns handle to the pending download.
+    """
+    mc.log.info(f"--- start download {file_path} for {project_path}")
+    project_info = mc.project_info(project_path, version=version)
+
+    mc.log.info(f"Got project info. version {project_info['version']}")
+
+    # set temporary directory and make sure the destination directory exists
+    temp_dir = os.path.join(tempfile.gettempdir(), "mergin_temp")
```

Comment:
this is not safe way of creating temporary dirs - it can be an issue if multiple calls to this function are done on one machine. Probably better to use tempfile.TemporaryDirectory (or directly mkdtemp)